shazam/shazam.py

```
@@ -0,0 +1,159 @@
+
+        if not urls:
+            return
+
+        async with message.channel.typing():
+            for url in urls:
+                try:
+                    media: bytes = await self.__aio_get(url)
+                    
+                    # Check if the file is a video and extract audio if necessary
+                    if 'video/' in url:
+                        media = await self.extract_audio_from_video(ctx, media)
```

Ruff (0.8.2)

Undefined name `ctx`

(F821)